apps/io-func-wallet-solution-backend/src/infra/http/handlers/health.ts

```diff
@@ -0,0 +1,59 @@
+import { pipe, identity } from "fp-ts/function";
+import * as T from "fp-ts/Task";
+import * as TE from "fp-ts/TaskEither";
+import * as RTE from "fp-ts/ReaderTaskEither";
+import * as RA from "fp-ts/ReadonlyArray";
+import { CosmosClient } from "@azure/cosmos";
+import * as O from "fp-ts/Option";
+import * as H from "@pagopa/handler-kit";
+import { logErrorAndReturnResponse } from "../utils";
+import {
+  PdvTokenizerHealthCheck,
+  getPdvTokenizerHealth,
+} from "@/infra/pdv-tokenizer/health-check";
+import { getCosmosHealth } from "@/infra/azure/cosmos/health-check";
+
+class HealthCheckError extends Error {
+  name = "HealthCheckError";
+  constructor(cause?: string) {
+    super(`The function is not healthy. ${cause}`);
+  }
+}
+
+const getHealthCheck: RTE.ReaderTaskEither<
+  {
+    cosmosClient: CosmosClient;
+    pdvTokenizerClient: PdvTokenizerHealthCheck;
+  },
+  Error,
+  void
+> = ({ cosmosClient, pdvTokenizerClient }) =>
+  pipe(
+    [
+      pipe({ cosmosClient }, getCosmosHealth),
+      pipe({ pdvTokenizerClient }, getPdvTokenizerHealth),
+    ],
+    RA.wilt(T.ApplicativePar)(identity),
+    T.chain(({ left }) =>
```

Comment:
what does this code do? (maybe a comment would help here)

Reply:
added a comment

Reply:
now?